utils/call-gen/src/create_program.rs

```diff
@@ -0,0 +1,88 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Create program call generator.
+
+use crate::{CallGenRng, GearCall, Seed};
+use gear_core::ids::CodeId;
+use gear_utils::{NonEmpty, RingGet};
+
+// code id, salt, payload, gas limit, value
+type CreateProgramArgsInner = (CodeId, Vec<u8>, Vec<u8>, u64, u128);
+
+/// Crate program args
+///
+/// Main type used to generate arguments for the `pallet_gear::Pallet::<T>::create_program` call.
+pub struct CreateProgramArgs(pub CreateProgramArgsInner);
+
+impl From<CreateProgramArgs> for CreateProgramArgsInner {
+    fn from(
+        CreateProgramArgs((code_id, salt, payload, gas_limit, value)): CreateProgramArgs,
```

Comment:
destructuring in functions signature looks too noisy. I would suggest to use `let ...` inside the body

Reply:
done